src/js/player.js

```diff
@@ -1530,7 +1535,12 @@ class Player extends Component {
    */
   techCall_(method, arg) {
     // If it's not ready yet, call method when it is
+    if (method in middleware.allowedSetters) {
+      return middleware.set(this.middleware_, this.tech_, method, arg);
+    }
+
```

Comment:
seems like we should check do this here, to prevent checking `this.tech_` in two places below.

```js
if (!this.tech_) {
  return
}  
```

Also it seems like `this.tech_.ready()` already checks if the tech is ready? Seems like we could merge a lot of this code together to make it much simpler.